src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/CVTextMouseActionData.cs

```diff
@@ -0,0 +1,544 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using JetBrains.Annotations;
+using RegressionGames.StateRecorder.BotSegments.JsonConverters;
+using RegressionGames.StateRecorder.BotSegments.Models.CVService;
+using RegressionGames.StateRecorder.JsonConverters;
+using RegressionGames.StateRecorder.Models;
+using UnityEngine;
+
+namespace RegressionGames.StateRecorder.BotSegments.Models
+{
+    /**
+     * <summary>Data for clicking on or moving the mouse to a CV Text location in the scene.
+     * This processes the results by finding the smallest bounding rect of the matching texts possible withinRect if specified and clicking on the center of that rect.</summary>
+     */
+    [Serializable]
+    public class CVTextMouseActionData : IBotActionData
+    {
+        // api version for this object, update if object format changes
+        public int apiVersion = SdkApiVersion.VERSION_14;
+
+        [NonSerialized]
+        public static readonly BotActionType Type = BotActionType.Mouse_CVText;
+
+        public string text;
+        public TextMatchingRule textMatchingRule = TextMatchingRule.Matches;
+        public TextCaseRule textCaseRule = TextCaseRule.Matches;
+
+        /**
+         * optionally limit the rect area where the TextData can be detected
+         */
+        [CanBeNull]
+        public CVWithinRect withinRect;
+
+        /**
+         * the list of actions to perform at this location
+         * Note: This does NOT re-evaluate the position of the CVText on each action.
+         * This is modeled as a list to allow you to click and/or un-click buttons on this Text location without re-evaluating the CV text match in between if you don't want to.
+         * If you want to validate a mouse visual effect in your criteria, and then mouse up in the next bot segment; that is also a normal way to use this system with just 1 list action per segment.
+         *
+         * If you want to perform click and drag mouse movements, you cannot and should not try to do that with this list.  You need to create 2 separate bot segments.
+         * Mouse down in one bot segment, then mouse up as a separate bot segment.
```

Comment:
Why is this the case? Is it because these actions don't include timings like the other mouse movement operations?

Reply:
This action occurs on the image found.  The only way to drag would be to give a new location somewhere else.

That somewhere else needs to be a different coordinate.